File: test_python_project/src/core_module/views/reports.py
from rest_framework import viewsets
from core_module.actions.add_task_to_list import AddTaskToList
from export_module.actions.perform_export import PerformExport
from export_module.tasks.run_exports import task_ExportAllTasks
from django.db import transaction


class ReportsViewSet(viewsets.ModelViewSet):
    def create(self):
        return AddTaskToList().execute()

    def export(self):
        sig = task_ExportAllTasks.si()
        transaction.on_commit(sig.apply_async)
        return PerformExport().execute()

    def total_amount(self):
        AddTaskToList()
        PerformExport()
        task_ExportAllTasks()
        pass

File: test_typescript_project/src/core_module/views/reports.ts
import { Hono } from "hono";
import { GetTasksList } from "@/core_module/actions/get_tasks_list";

export const reportsRouter = new Hono().get("/reports/tasks", async (c) => {
  const { tasks } = await new GetTasksList().execute();
  return c.json({ total: tasks.length });
});

